tools/build_h3_tools.sh

```diff
@@ -68,13 +72,59 @@ else
   num_threads=$(sysctl -n hw.logicalcpu)
 fi
 
+# boringssl
+echo "Building boringssl..."
+
+# We need this go version.
+if [ ! -d ${BASE}/go ]; then
+  sudo mkdir ${BASE}/go
+fi
+
+if [ `uname -m` = "arm64" ]; then
+    ARCH="arm64"
+else
+    ARCH="amd64"
+fi
+
+if [ `uname -s` = "Darwin" ]; then
+    OS="darwin"
+else
+    OS="linux"
+fi
+
+wget https://go.dev/dl/go1.20.1.${OS}-${ARCH}.tar.gz
+sudo rm -rf ${BASE}/go && sudo tar -C ${BASE} -xf go1.20.1.${OS}-${ARCH}.tar.gz
+rm go1.20.1.${OS}-${ARCH}.tar.gz
+
+GO_BINARY_PATH=${BASE}/go/bin/go
+if [ ! -d boringssl ]; then
+  git clone https://boringssl.googlesource.com/boringssl
+  cd boringssl
+  git checkout 31bad2514d21f6207f3925ba56754611c462a873
+  cd ..
+fi
+cd boringssl
+if [ ! -d build ]; then
+  mkdir build
+fi
+cd build
+cmake \
+  -DGO_EXECUTABLE=${GO_BINARY_PATH} \
+  -DCMAKE_INSTALL_PREFIX=${BASE}/boringssl \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DBUILD_SHARED_LIBS=1 ../
+
+${MAKE} -j ${num_threads}
+sudo ${MAKE} install
+cd ..
+
 # Build quiche
 # Steps borrowed from: https://github.com/apache/trafficserver-ci/blob/main/docker/rockylinux8/Dockerfile
 echo "Building quiche"
-QUICHE_BASE=${BASE:-"/opt/quiche"}
+if [ -z ${BASE+x} ]; then QUICHE_BASE="/opt/quiche"; else QUICHE_BASE=${BASE}/quiche; fi
```

Comment:
I had to make this so we either use ${BASE}/quiche(if base is passed) or /opt/quiche.

Reply:
I don't know why this got posted twice, but oh well. :)